runtime/doc/lua-plugin.txt

```diff
@@ -0,0 +1,433 @@
+*lua-plugin.txt*                     Nvim
+
+                            NVIM REFERENCE MANUAL
+
+		  Guide to developing Lua plugins for Nvim
+
+
+                                       Type |gO| to see the table of contents.
+
+==============================================================================
+Introduction                                                       *lua-plugin*
+
+This is a guide for getting started with Nvim plugin development. It is not
```

Comment:
Reading this I got the impression that this is more an intermediate best practice guide and not a beginner getting started guide.  
(E.g compare with the `:h write-plugin` intro)

Reply:
How about  I remove the "getting started with"?

Reply:
Let's change the content to be beginner friendly instead with a way to "advance" to the more intermediate topics.

Reply:
I'll rereview it when I have more time with more useful comments.

Reply:
I'm not sure if just removing "getting started" will cut it.

Overall this feels more like `plugin-ui-patterns` than `lua-plugin`.

It talks about:

- Best practices in regards to keymap definitions for users
    - Should tie into `lua-guide-mappings`
- Best practices on how to expose configuration options to users
- Best practices on how to design command APIs
    - Should tie into `lua-guide-commands`
    
- Performance considerations
    - Partly covered by `lua-guide-modules` and `lua-module-load` and also related to `autoload`

None of this is beginner content for how to get a plugin working.
And many parts are not exclusive to Lua.

What I'd expect from a "How to write a plugin" targeted towards beginner is:

- Language choices and link to resources to learn them (vim, lua, but also external providers)
- Entry points (runtime path, packages, plugin/, ftplugin, ftdetect, autocmd hooks)
- Available APIs

But a lot of this is covered via `lua-guide` and the various vim resources. I also suspect that writing something like this is not what Justin had in mind, but rather something that's closer to the style of `:h dev-patterns`


But let's wait for @justinmk to clarify what his intentions were.